src/vs/platform/terminal/common/environmentVariableCollection.ts

```diff
@@ -28,6 +33,15 @@ export class MergedEnvironmentVariableCollection implements IMergedEnvironmentVa
 			while (!next.done) {
 				const mutator = next.value[1];
 				const key = next.value[0];
+
+				// Only Python env extension can modify Python activate env var.
+				if (PYTHON_ACTIVATION_VARS_PATTERN.test(key) &&
+					PYTHON_ENV_EXTENSION_ID !== extensionIdentifier) {
+					this._logService.warn(`Extension '${extensionIdentifier}' attempted to modify Python activation variable '${key}' but was blocked`);
```

Comment:
ended up touching so many files, to have logs here to denote in case other extension where trying to modify Python activate vars.

Honestly not sure if this is worth all the changes.